### PR TITLE
Move HttpContextAccessor to HttpAbstractions

### PR DESCRIPTION
**Pull request by HaoK**
cc @davidfowl @Tratcher

### CODE REVIEW

File: src/Microsoft.AspNet.Http.Abstractions/HttpContextAccessor.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+using System;
+#if DNX451
+using System.Runtime.Remoting.Messaging;
+using System.Runtime.Remoting;
+#elif DNXCORE50
+using System.Threading;
+#endif
+
+namespace Microsoft.AspNet.Http
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Http.Abstractions/HttpContextAccessor.cs]: Http.Internal?

**Comment by Tratcher** [comment]
:shipit:

File: src/Microsoft.AspNet.Http.Abstractions/HttpContextAccessor.cs (hunk 1 above)
Comment (HaoK) [inline src/Microsoft.AspNet.Http.Abstractions/HttpContextAccessor.cs]: Hosting is going to need to use this still, but I don't know what the guidance is on the internal namespace.

Comment (HaoK) [inline src/Microsoft.AspNet.Http.Abstractions/HttpContextAccessor.cs]: Seems a little weird to put it in Internal when its another Repo using it though?

Comment (Tratcher) [inline src/Microsoft.AspNet.Http.Abstractions/HttpContextAccessor.cs]: The internal namespace is for framework infrastructure that's probably not used by the general public.
Compare to DefaultHttpContext which Hosting also uses: https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNet.Http/DefaultHttpContext.cs#L15

Comment (HaoK) [inline src/Microsoft.AspNet.Http.Abstractions/HttpContextAccessor.cs]: Sounds reasonable, but there's nothing else in Internal in HttpAbstractions currently, but I can create the folder and move it there if you think its the right thing do

**Comment by HaoK** [comment]
Updated PR moving implementation to internal namespace, I didn't create a folder since there is no folder in the other projects in Http currently

File: src/Microsoft.AspNet.Http.Abstractions/HttpContextAccessor.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+using System;
+#if DNX451
+using System.Runtime.Remoting.Messaging;
+using System.Runtime.Remoting;
+#elif DNXCORE50
+using System.Threading;
+#endif
+
+namespace Microsoft.AspNet.Http.Internal
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Http.Abstractions/HttpContextAccessor.cs]: Oh, I just realized this was in the Abstractions package. This implementation should move to the Http package.

Comment (HaoK) [inline src/Microsoft.AspNet.Http.Abstractions/HttpContextAccessor.cs]: Ok that makes things feel much nicer :)

**Comment by Tratcher** [comment]
:shipit: :shipit:

**Pull request closed without merging**